migrations/20201009072158-UniquePlayerUserIdConstraint.ts

```diff
@@ -0,0 +1,62 @@
+import Base from "db-migrate-base";
+
+/**
+ * Add the UNIQUE constraint to the user_id column of players.
+ */
+export async function up(
+  db: Base,
+  callback: Base.CallbackFunction
+): Promise<void> {
+  db.changeColumn(
+    "players",
+    "user_id",
+    {
+      type: "int",
+      notNull: true,
+      unique: true,
+      foreignKey: {
+        name: "fk_players_user_id",
```

Comment:
?

Reply:
i'm honestly confused on if this is supposed to be a placeholder but please replace it if it is :")

Reply:
`fk` is short for "foreign key" here, and I'm creating a unique name for the foreign key constraint. I was using a bit of a mix of conventions that I've seen here: https://stackoverflow.com/questions/2240929/what-is-a-proper-naming-convention-for-mysql-fks/25336979

I used `fk_<table name>_<referencing column>` in this case. We could opt for another convention if necessary! I think the constraint name is not used in any place in our application code (and is sometimes autogenerated by other ORMs), it just needs to be a unique key.

Reply:
got it!! thank you for the context, it helped a lot since i wasn't sure if I was interpreting it wrong. I think it should be fine to maintain convention and use fk to represent foreign keys